packages/orchestrator-ui-components/src/components/WfoWorkflowSteps/WfoWorkflowStepList/WfoWorkflowStepList.tsx

```diff
@@ -35,18 +48,50 @@ export const WfoWorkflowStepList = React.forwardRef(
         const initialStepListItems: StepListItem[] = steps.map((step) => ({
             step,
             isExpanded: false,
-        }));
+        })); // If the step is in the suspend state, we show the user input form
 
         const [stepListItems, setStepListItems] =
             useState(initialStepListItems);
 
-        useEffect(
-            () =>
-                setStepListItems((previousStepListItems) =>
-                    updateStepListItems(previousStepListItems, steps),
+        const persistStepListItemState = (
```

Comment:
Why moving logic from utils in to the component?

Reply:
Because it's specific to this component and there was one local updateStepListItem and a updateStepListItems in the utils which was confusing to me.